synapse/storage/databases/main/room.py

```diff
@@ -1809,6 +1827,36 @@ def _store_partial_state_room_txn(
         )
         self._invalidate_cache_and_stream(txn, self.is_partial_state_room, (room_id,))
 
+    async def write_partial_state_rooms_join_event_id(
+        self,
+        room_id: str,
+        join_event_id: str,
+    ) -> None:
+        """Record the join event which resulted from a partial join.
+
+        We do this separately to `store_partial_state_room` because we need to wait for
+        the join event to be persisted. Otherwise we violate a foreign key constraint.
+        """
+        await self.db_pool.runInteraction(
+            "write_partial_state_rooms_join_event_id",
+            self._store_partial_state_room_txn,
```

Comment:
Also wrong function

Reply:
oh bugger. Why didn't mypy spot that?

Reply:
Because we don't use ParamSpec in synapse.storage.database.py. Will refrain for now but I've found my next brain cleaner.